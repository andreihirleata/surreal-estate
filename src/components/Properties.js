import React, { useEffect, useState } from "react";
import { useLocation } from "react-router-dom";
import PropertyCard from "./PropertyCard";
import axios from "axios";
import "../styles/Properties.css";
import Sidebar from "./Sidebar";

const Properties = ({ userId }) => {
  const [properties, setProperties] = useState([]);
  const { search } = useLocation();
  useEffect(() => {
    axios
      .get(`http://localhost:4000/api/v1/PropertyListing${search}`)
      .then((res) => setProperties(res.data))
      .catch((err) => {
        console.log(err);
      });
  }, [search]);

  const handleSaveProperty = (propertyId) => {
    axios.post("http://localhost:4000/api/v1/Favourite", {
      propertyListing: propertyId,
      fbUserId: userId,
    });
  };

  return (
    <div className="properties">
      <Sidebar className="sidebar-wrapper" />
      <ul className="property-grid">
        {properties.map((property) => (
          <li className="property-card" key={property._id}>
            <PropertyCard
              {...property}
              userId={userId}
              onSaveProperty={handleSaveProperty}
            />
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Properties;
