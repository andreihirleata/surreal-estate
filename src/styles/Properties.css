.property-grid {
  min-height: 100vh;
  padding: 10px;
  display: flex;
  flex-wrap: wrap;
  background-color: #e5e5e5;
  margin: 0;
  flex-direction: column;
  justify-content: center;
  padding-top: 3rem;
}

.property-grid > li {
  margin: 10px 20px;
  padding: 15px;
  height: 360px;
  background-color: white;
  min-width: 100px;
  text-align: center;
  font-weight: 800;
  margin-bottom: 1rem;
  border-radius: 9px;
  box-shadow: 0 10px 20px rgba(0, 0, 0, 0.19), 0 6px 6px rgba(0, 0, 0, 0.23);
}

.sidebar-wrapper {
  margin-bottom: 5rem;
}

@media (min-width: 600px) {
  .property-grid {
    flex-direction: row;
  }
}
