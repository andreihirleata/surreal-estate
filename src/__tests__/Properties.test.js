import React from "react";
import { render } from "@testing-library/react";
import Properties from "../components/Properties";

test("renders properties component", () => {
  const { asFragment } = render(<Properties />);
  const component = asFragment();
  expect(component).toMatchSnapshot();
});

test("renders Properties Page text", () => {
  const { getByText } = render(<Properties />);
  expect(getByText(/properties page/i)).toBeInTheDocument();
});
