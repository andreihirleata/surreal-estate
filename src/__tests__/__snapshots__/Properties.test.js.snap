// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properties component 1`] = `
<DocumentFragment>
  <div>
    <p>
      Properties Page
    </p>
  </div>
</DocumentFragment>
`;
